tls_codec/src/quic_vec.rs

```diff
@@ -127,8 +127,9 @@ pub fn write_variable_length(content_length: usize) -> Result<Vec<u8>, Error> {
         }
     }
     let mut len = content_length;
-    for b in length_bytes.iter_mut().rev() {
-        *b |= (len & 0xFF) as u8;
+    let l = length_bytes.len();
```

Comment:
You listed the hax issues on the PR. Can you add them to each of the changes as well? Then it's easier to understand why it has to be this way, and we can find all places later when we fix the hax issues.